### PR TITLE
[4.0] Array to php 5.4+ notation [Plugins 2: Editors, Editors-XTD, Extension and Fields]

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Improvement.

### Summary of Changes

Since 4.0 doesn't support php 5.3 anymore, use php 5.4+ array notation in Editors, Editors-XTD, Extension and Fields plugins.

### Testing Instructions

Simple code review.

### Documentation Changes Required

None.

### CODE REVIEW

File: plugins/editors/codemirror/codemirror.php [hunk 1]
```diff
@@ -63,9 +63,9 @@ public function onInit()
 		JPluginHelper::importPlugin('editors_codemirror');
 
 		// At this point, params can be modified by a plugin before going to the layout renderer.
-		JFactory::getApplication()->triggerEvent('onCodeMirrorBeforeInit', array(&$this->params));
+		JFactory::getApplication()->triggerEvent('onCodeMirrorBeforeInit', [&$this->params]);
 
-		$displayData = (object) array('params'  => $this->params);
+		$displayData = (object) ['params'  => $this->params];
```
Comment (Quy) [inline plugins/editors/codemirror/codemirror.php]: Remove extra space `['params'  =>`

Comment (andrepereiradasilva) [inline plugins/editors/codemirror/codemirror.php]: space removed

File: plugins/editors/tinymce/tinymce.php [hunk 2]
```diff
@@ -1166,13 +1166,13 @@ public static function getToolbarPreset()
 				'hr', 'table', '|',
 				'subscript', 'superscript', '|',
 				'charmap', 'pastetext' , 'preview'
-			),
-			'toolbar2' => array(),
+			],
+			'toolbar2' => [],
 		);
```
Comment (frankmayer) [inline plugins/editors/tinymce/tinymce.php]: Should be `];` (Line 1171)

**Review by frankmayer — CHANGES_REQUESTED** [review]

File: plugins/editors/tinymce/tinymce.php [hunk 3]
```diff
@@ -1191,8 +1191,8 @@ public static function getToolbarPreset()
 				'cut', 'copy', 'paste', 'pastetext', '|',
 				'visualchars', 'visualblocks', 'nonbreaking', 'blockquote', 'template', '|',
 				'print', 'preview', 'codesample', 'insertdatetime', 'removeformat',
-			),
-			'toolbar2' => array(),
+			],
+			'toolbar2' => [],
 		);
```
Comment (frankmayer) [inline plugins/editors/tinymce/tinymce.php]: Should be ]; (Line 1196)

**Review by frankmayer — CHANGES_REQUESTED** [review]

File: plugins/editors/tinymce/tinymce.php (hunk 2 above)
Comment (andrepereiradasilva) [inline plugins/editors/tinymce/tinymce.php]: thanks corrected both

**Comment by frankmayer** [comment]
I have tested this item :white_check_mark: successfully on da68252f30b2fef3aa1d05f070803b928a765b84<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16653">issues.joomla.org/tracker/joomla-cms/16653</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on da68252f30b2fef3aa1d05f070803b928a765b84<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16653">issues.joomla.org/tracker/joomla-cms/16653</a>.</sub>

**Pull request closed without merging**